rename.cc

```diff
@@ -0,0 +1,54 @@
+#include <fcntl.h>
+#include <sys/stat.h>
+
+#include "./capsicum-test.h"
+#define RENAME_TEST_FILENAME        ".rename_test"
+#define RENAMEAT_TEST_FILENAME      ".renameat_test"
+#define RENAMEAT_TEST_DIR           ".renameat_testdir"
+
+/*
+added to test the renameat syscall for the case that
+    - the "to" file already exists
+    - the "to" file is specified by an absolute path
+    - the "to" file descriptor is used
+
+details at: https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=222258
+*/
+
+const char * create_tmp_src(const char* filename) {
+    const char *src_path = TmpFile(filename);
+    int src_fd = open(src_path, O_CREAT|O_RDWR, 0644);
+    close(src_fd);
+    return src_path;
+}
+
+TEST(Rename, rename_abs) {
+    const char *src_path = create_tmp_src(RENAME_TEST_FILENAME);
+    int ret = rename(src_path, src_path);
+    EXPECT_OK(ret);
```

Comment:
(nit) For consistency with all the other tests, combine into one line:
```c
    EXPECT_OK(rename(src_path, src_path));
```
(Passim)